app/src/main/java/com/example/chrono/util/adapters/CircuitViewAdapter.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.chrono.util.adapters
+
+import android.annotation.SuppressLint
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.chrono.R
+import com.example.chrono.util.objects.CircuitObject
+import com.google.android.material.textview.MaterialTextView
+
+class CircuitViewAdapter(private val data: List<CircuitObject>) :
+    RecyclerView.Adapter<CircuitViewAdapter.CircuitViewHolder>() {
+
+    class CircuitViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val name: MaterialTextView = view.findViewById(R.id.circuit_name)
+        val numSets: MaterialTextView = view.findViewById(R.id.num_sets)
+        val timeRest: MaterialTextView = view.findViewById(R.id.rest_time)
+        val timeWork: MaterialTextView = view.findViewById(R.id.work_time)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CircuitViewHolder {
+        val adapterLayout =
+            LayoutInflater.from(parent.context).inflate(R.layout.circuit_card_lay, parent, false)
+        return CircuitViewHolder(adapterLayout)
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onBindViewHolder(holder: CircuitViewHolder, position: Int) {
+        val circuit = data[position]
+        holder.name.text = circuit.name.toString()
+        holder.numSets.text = circuit.sets.toString() + " "
```

Comment:
" " at the start?

Reply:
Not entirely sure what you mean here, the space is between the number of sets and then the word "Sets" which is hardcoded in anyways. Wait as I'm typing this I just realized I have a dumb implementation for this hold up.